inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/source/JdbcSource.java

```diff
@@ -0,0 +1,306 @@
+        statCycle.setStartTime(dateFormat.format(calendar.getTime()));
+
+        calendar.set(Calendar.HOUR_OF_DAY, 23);
+        calendar.set(Calendar.MINUTE, 59);
+        statCycle.setEndTime(dateFormat.format(calendar.getTime()));
+        return new ArrayList<StartEndTime>() {
+
+            {
+                add(statCycle);
+            }
+        };
+    }
+
+    /**
+     * Create data source
+     */
+    protected void createDataSource() {
+        HikariConfig config = new HikariConfig();
+        config.setDriverClassName(sourceConfig.getDriverClassName());
+        config.setJdbcUrl(sourceConfig.getJdbcUrl());
+        config.setUsername(sourceConfig.getUsername());
+        config.setPassword(sourceConfig.getPassword());
+        config.setConnectionTimeout(Configuration.getInstance().get(KEY_DATASOURCE_CONNECTION_TIMEOUT,
+                DEFAULT_CONNECTION_TIMEOUT));
+        config.addDataSourceProperty(CACHE_PREP_STMTS,
+                Configuration.getInstance().get(KEY_CACHE_PREP_STMTS, DEFAULT_CACHE_PREP_STMTS));
+        config.addDataSourceProperty(PREP_STMT_CACHE_SIZE,
+                Configuration.getInstance().get(KEY_PREP_STMT_CACHE_SIZE, DEFAULT_PREP_STMT_CACHE_SIZE));
+        config.addDataSourceProperty(PREP_STMT_CACHE_SQL_LIMIT,
+                Configuration.getInstance().get(KEY_PREP_STMT_CACHE_SQL_LIMIT, DEFAULT_PREP_STMT_CACHE_SQL_LIMIT));
+        config.setMaximumPoolSize(
+                Configuration.getInstance().get(KEY_DATASOURCE_POOL_SIZE,
+                        DEFAULT_DATASOURCE_POOL_SIZE));
+        dataSource = new HikariDataSource(config);
+    }
+
+    /**
+     * Destory
+     */
+    public void destroy() {
+        for (Map.Entry<Integer, ScheduledExecutorService> timer : statTimers.entrySet()) {
+            timer.getValue().shutdown();
+        }
+    }
+
+    /**
+     * Stat server
+     */
+    class StatServer implements Runnable {
+
+        private final int statBackTimes;
+
+        public StatServer(int statBackTimes) {
+            this.statBackTimes = statBackTimes;
+        }
+
+        public void run() {
+            long currentTimestamp = System.currentTimeMillis();
+            LOG.info("Stat source data at {},stat back times:{}", currentTimestamp, statBackTimes);
+
+            statByStep();
+
+            long timeCost = System.currentTimeMillis() - currentTimestamp;
+            LOG.info("Stat source data cost time:{}ms,stat back times:{}", timeCost, statBackTimes);
+        }
+
+        /**
+         * Stat by step
+         */
+        public void statByStep() {
+            List<CompletableFuture<String>> futures = new ArrayList<>();
+            for (String auditId : auditIds) {
+                CompletableFuture<String> future = CompletableFuture.supplyAsync(() -> {
+                    aggregate(auditId);
+                    return "Task " + auditId + " completed";
+                });
+                futures.add(future);
+            }
```

Comment:
1. please  do the empty check for `auditIds` first
2. The future for each auditId does not need return a string, CompletableFuture<Void> is enough